### PR TITLE
Add docker image based on timescale/timescaledb-ha

### PR DESCRIPTION
**Pull request by JamesGuthrie**
We would like to use a timescaledb-ha based docker image for Promscale's
end-to-end tests, with the most recent (development) version of the
Promscale extension.

This change prepares such a docker image for each opened pull request
and merge to the develop and master branches, pushing the image to the
`ghcr.io/timescale/dev_promscale_extension` registry with a version tag.

For PRs the version tag is derived from the source branch name, but
replacing special characters (e.g. '/') with '-'. For merges to develop
and master, the version tag is the branch name.

This change also modifies the alpine docker build and pushes it to the
registry (to improve caching). Alpine images are pushed with the suffix
`-alpine`.

Example version tags are:
- `jg-ha-dockerfile-ts2-pg14` (based on the `jg/ha-dockerfile branch`)
- `jg-ha-dockerfile-ts2-pg14-alpine` (same as above, but alpine)
- `develop-ts2-pg14`
- `master-ts2-pg14`

### CODE REVIEW

**Review by jgpruitt — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -135,13 +135,15 @@ release-test: release-tester ## Test the currently selected release package
...
-docker-image-build-12 docker-image-build-13 docker-image-build-14: Dockerfile $(SQL_FILES) $(SRCS) Cargo.toml Cargo.lock $(RUST_SRCS)
+docker-image-build-12 docker-image-build-13 docker-image-build-14: alpine.Dockerfile $(SQL_FILES) $(SRCS) Cargo.toml Cargo.lock $(RUST_SRCS)
 	docker buildx build $(if $(PUSH),--push,--load) \
 		--build-arg TIMESCALEDB_VERSION=$(TIMESCALEDB_VER) \
 		--build-arg PG_VERSION_TAG=$(PG_BUILD_VER) \
+		-t local/dev_promscale_extension:head-ts2-$(PG_BUILD_VER) \
 		-t $(IMAGE_NAME):$(EXT_VERSION)-$(TIMESCALEDB_VER)-$(PG_BUILD_VER) \
 		-t $(IMAGE_NAME):$(EXT_VERSION)-ts$(TIMESCALEDB_MAJOR)-$(PG_BUILD_VER) \
 		-t $(IMAGE_NAME):latest-ts$(TIMESCALEDB_MAJOR)-$(PG_BUILD_VER) \
+        -f alpine.Dockerfile \
```
Comment (JamesGuthrie) [inline Makefile:146]: Note: This should transition to the ha.Dockerfile image, _but_ it doesn't build on M1 macs at the moment.

**Review by sumerman — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (JamesGuthrie) [reply Makefile:146]: I've opened https://github.com/timescale/promscale_extension/issues/140 to track

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -7,23 +7,73 @@ on:
...
+            TIMESCALEDB_VER=${{ matrix.tsversion }}
+          context: .
+          file: ${{matrix.base}}.Dockerfile
+          push: true
+          tags: ghcr.io/timescale/dev_promscale_extension:${{steps.metadata.outputs.branch_name}}-ts${{steps.metadata.outputs.tsmajor}}-pg${{matrix.pgversion}}${{steps.metadata.outputs.build_type_suffix}}
+          # Note: it's necessary to use a different cache scope to achieve caching for both Ubuntu and Alpine images
+          cache-from: type=gha,scope=${{matrix.base}}
+          cache-to: type=gha,mode=max,scope=${{matrix.base}}
+
+  # Added to summarize the status of docker builds (otherwise we need keep this up to date in branch protection rules)
```
Comment (sumerman) [inline .github/workflows/docker.yaml:66]: IMO the commit message wording better conveys the intent. I suggest using the same text in the comment.

**Review by sumerman — APPROVED** [review]

File: .github/workflows/docker.yaml (hunk 2 above)
Comment (JamesGuthrie) [reply .github/workflows/docker.yaml]: Thank you! You're right. I've updated this.

**Pull request merged**